qhana_plugin_registry/db/models/templates.py

```diff
@@ -231,6 +267,34 @@ class TemplateTab(IdMixin, ExistsMixin, NameDescriptionMixin):
     def plugin_filter(self) -> dict:
         return json.loads(self.filter_string)
 
+    @classmethod
+    def get_by_name(cls, name: str) -> "Optional[TemplateTab]":
+        q = select(cls).filter(cls.name == name)
+        found_tab: Optional[TemplateTab] = DB.session.execute(q).scalar_one_or_none()
+        return found_tab
+
+    @classmethod
+    def get_or_create(cls, tab: dict) -> "TemplateTab":
```

Comment:
just pass the template as a second param to this function (also for `get_or_create_all`)